src/main/java/org/apache/maven/plugins/javadoc/AbstractJavadocMojo.java

```diff
@@ -6879,6 +6892,33 @@ private void writeDebugJavadocScript( String cmdLine, File javadocOutputDirector
         }
     }
 
+    /**
+     * Write a files containing the javadoc errors and warnings.
+     *
+     * @param errorsAndWarnings      the javadoc errors and warnings as string, not null.
+     * @param javadocOutputDirectory the output dir, not null.
+     * @since 3.4.2-SNAPSHOT
+     */
+    private void writeErrorFile( String errorsAndWarnings, File javadocOutputDirectory )
+    {
+        File commandLineFile = new File( javadocOutputDirectory, ERRORS_FILE_NAME );
+        commandLineFile.getParentFile().mkdirs();
+
+        try
+        {
+            FileUtils.fileWrite( commandLineFile.getAbsolutePath(), null /* platform encoding */, errorsAndWarnings );
+
+            if ( !SystemUtils.IS_OS_WINDOWS )
+            {
+                Runtime.getRuntime().exec( new String[]{ "chmod", "a+x", commandLineFile.getAbsolutePath() } );
```

Comment:
Why is this necessary?

Reply:
Good point, I was following this pattern found elsewhere in the file:

https://github.com/apache/maven-javadoc-plugin/blob/master/src/main/java/org/apache/maven/plugins/javadoc/AbstractJavadocMojo.java#L6925

But after taking a second look at it it's meant to be an executable file. I'll remove this entire if block.

Reply:
The file referenced is a script, totally unrelated to this.

Reply:
Yeah, now this simply just does file writing with no modifications to the permissions.